### DISCUSSION TITLE
Make animation using simple.XMLMultiBlockDataReader or vtkXMLMultiBlockDataReader

### DISCUSSION THREAD

**Discussion by yjtwinpark** [Kitware/trame#210] [Q&A] [answered]
I want load vtm file, which has time series and multi block.
Structure like follows.
   FileName_0.vtm
         FileName_0_0_0.vtu, Filename_0_1_0.vtu, ....
   FileName_1.vtm
         FileName_1_0_0.vtu, Filename_1_1_0.vtu, ....
Time value and filename is in vtm file, and information of each point, cell... are in vtu file.
(this is just I import  d3plot file, and export as vtm file in local paraview app.)

When I read this, i used XMLMultiBlockDataReader, and it works well.
And I made animation function too, refer to "Time Animation Example"

But I have one problem, I want to hide all block and show only chosen block 
simple.ShowAll and simple.HideAll work well, because all multiblocks are included, 
But simple.show or simple.hide is not work well.
They require "source", but when I use XMLMultiBlockDataReader, 
both reader (simple.XMLMultiBlockDataReader(FileName=dataNameList))  and
output (reader.GetClientSideObject().GetOutputDataObject(0)) not worked for hide and show

In the case of vtkXMLMultiBlockDataReader, it cannot open vtm file as list,
but there is no class or function for open like simple.XMLMultiBlockDataReader(FileName=dataNameList).
and therefore I cant use simple.GetAnimationScene for vtkXMLMultiBlockDataReader

Shortly, please suggest me how to...
(1) hide and show one block if i use XMLMultiBlockDataReader to open vtm file
(2) make animation if i use vtkXMLMultiBlockDataReader

please help me.

thank you.

**Comment by jourdain** [comment, marked as answer]
Since you are using ParaView, you might be able to use the trace to extract the Python section of a UI action (use multi-block inspector). 
Otherwise, you should ask that Python/ParaView question on [ParaView discourse](https://discourse.paraview.org/).

**Reply by yjtwinpark** [reply]
I solved! In fact, I followed the trane Python script using Paraview, but I had problem, because there was misspell BlockSelectors as BlockSelector. (My Python Interpreter cant recognize function BlockSelectors) Thank you for your help!
